Add choropleth incidence rate map to index page

RobertRosca opened the pull request:
Adds in choropleth incidence rate map plot to the main index page.

Simplest approach for this was to add a few cells to the `tools/generate-individiual-plots.ipynb` notebook which generate and save the HTML plot, then added RST raw HTML include to the index page at `tools/pelican/content/pages/index.rst` to 'import' this plot to the index page.

Some issues remain, like the background for the plot not matching the background for the page, haven't looked at this much so the fix might be easy but I'm not too sure yet.

RobertRosca commented:
For whatever reason the staging page seems to not be loading, sometimes github pages updates can be a bit slow, here's a render of the index page: https://htmlpreview.github.io/?https://github.com/oscovida/staging/blob/master/195/index.html

kirienko commented:
Good start!
However, I suppose we can do even better if we subtly change several things:
- [ ]  as you already mentioned, background for the plot not matching the background for the page
- [ ] the colour map is a bit counter-intuitive: usually people expect something like red-yellow-green colours for risk vs safe areas, or a scale like vibrant colours — muted colours, or even black and white. Here we see yellow for the most dangerous areas while more red means more safe :man_shrugging: 
- [ ] the graph frame is quite small, though maybe if we solve the background colour issue this won't be looking like a problem
- [ ] also it'd be helpful to show control elements of the map (zoom in and out, reset etc) constantly. Otherwise it's not obvious that the map is interactive, unless you accidentally hover over it
- [ ] the content of the pop-ups is a bit strange, but I suppose it's something like a default values (I doubt people want to know `iso_alpha` values)

kirienko commented:
Also maybe we need to add an explanation to the graph below (I mean the random country we plot below the map)

fangohr commented:
Very cool, thank you.

In addition to Yury's feedback: if it is possible to have a URL associated with each country, so that you can click on it (maybe the URL can show in the hover window?) to be taken to the country, that would be cool. Don't know if this is possible.

RobertRosca commented:
> In addition to Yury's feedback: if it is possible to have a URL associated with each country, so that you can click on it (maybe the URL can show in the hover window?) to be taken to the country, that would be cool. Don't know if this is possible.

I'm afraid that's not possible right now with plotly: https://github.com/plotly/plotly.js/issues/998

RobertRosca commented:
> Good start!
> However, I suppose we can do even better if we subtly change several things:
> - [x]   as you already mentioned, background for the plot not matching the background for the page
> - [x]  the colour map is a bit counter-intuitive: usually people expect something like red-yellow-green colours for risk vs safe areas, or a scale like vibrant colours — muted colours, or even black and white. Here we see yellow for the most dangerous areas while more red means more safe man_shrugging
> - [x]  the content of the pop-ups is a bit strange, but I suppose it's something like a default values (I doubt people want to know `iso_alpha` values)
> - [x]  also it'd be helpful to show control elements of the map (zoom in and out, reset etc) constantly. Otherwise it's not obvious that the map is interactive, unless you accidentally hover over it
> - [x]  the graph frame is quite small, though maybe if we solve the background colour issue this won't be looking like a problem

Fixed most of these now. Only issue is that the control elements are supposed to fade in and out, enabling them all the time makes them clash with the legend, I'll look into fixing that tomorrow.

While testing it locally I realised that, at least imo, the page looks nicer without a lot of the CSS :sweat_smile:

![image](https://user-images.githubusercontent.com/32569096/101649663-711adf00-3a3b-11eb-860c-4f7b7de6d04d.png)

I prefer the cleaner look without the background/shadows but I guess a change like that should be in a different PR.

kirienko commented:
Wow, that's cool!

The only question, but a very important one, _what does this map show_? :worried:

RobertRosca commented:
> Wow, that's cool!
> 
> The only question, but a very important one, _what does this map show_? worried

It's on the legend, if it wasn't blocked by the controls :stuck_out_tongue: "14 Day Incidence Rate"

kirienko commented:
Yeah, I've realised already that this is exactly [this page](https://oscovida.github.io/14-day-incidence-rate.html)!

RobertRosca commented:
Yep. Same data used to colour the rows on the incident rate table.

fangohr commented:
It's really nice!

Some review items:

1. I would like to show the 7-day incidence here. It is the metric that is used more now, certainly in Germany and Europe, but I'd even argue internationally. And we already use the 7-day incidence as our main way of presenting data normalised by population (in https://oscovida.github.io/countries-incidence-rate.html). Can we change to that, please?

fangohr commented:
2. In the balloon hint that comes up when you hover the mouse, can we add "cases per week per 100,000" after the number that shows the incidence? That way, you would know what the number is.

fangohr commented:
3. As we have a page that displays these incidence numbers beautifully, I would add a caption to the figure (i.e. a line of html below the figure) that says something like: 

"The map shows [7-day incidence rates per 100,000 people](https://oscovida.github.io/countries-incidence-rate.html) as reported by the Johns-Hopkins University."

fangohr commented:
One more thing to consider (but could be after this has been merged): in the current way the colour scale is chosen, any outlier with a high number of cases will dominate the colour scheme for the others. Today we have Georgia with the highest number, with other countries like Italy, Germany, the UK and France looking positively tame and in good order. But the map should be quite red everywhere I think.

So a possible solution would be to assign the color at the top of the colour scale (deep red) not to the highest incidence, but to the country that has the n-th highest incidence (i.e. the 10th highest), and all countries with higher values are shown in the same deep red. The numerical details one gets from the table or when hovering. (I suppose that is similar to saying that all countries in the top 5%-tile are deep red.)

This may need some experimentation (and we can change it any time, as the colour will change every day anyway), but I think is worth doing. 

You could summarise my thoughts as: I think this should look more alarming than it is at the moment ... ;-)

RobertRosca commented:
> It's really nice!
> 
> Some review items:
> 
>     1. I would like to show the 7-day incidence here. It is the metric that is used more now, certainly in Germany and Europe, but I'd even argue internationally. And we already use the 7-day incidence as our main way of presenting data normalised by population (in https://oscovida.github.io/countries-incidence-rate.html). Can we change to that, please?

Changed it to 7 days now, that was an oversight on my part, sorry!

RobertRosca commented:
> 2. In the balloon hint that comes up when you hover the mouse, can we add "cases per week per 100,000" after the number that shows the incidence? That way, you would know what the number is.

I think that would take too much space up in the tooltip thing, it could go above the plot though?

RobertRosca commented:
> 3. As we have a page that displays these incidence numbers beautifully, I would add a caption to the figure (i.e. a line of html below the figure) that says something like:
> 
> "The map shows [7-day incidence rates per 100,000 people](https://oscovida.github.io/countries-incidence-rate.html) based on data reported by the [Johns-Hopkins University](https://oscovida.github.io/data-sources.html)."

Added that text in now

RobertRosca commented:
> One more thing to consider (but could be after this has been merged): in the current way the colour scale is chosen, any outlier with a high number of cases will dominate the colour scheme for the others. Today we have Georgia with the highest number, with other countries like Italy, Germany, the UK and France looking positively tame and in good order. But the map should be quite red everywhere I think.
> 
> So a possible solution would be to assign the color at the top of the colour scale (deep red) not to the highest incidence, but to the country that has the n-th highest incidence (i.e. the 10th highest), and all countries with higher values are shown in the same deep red. The numerical details one gets from the table or when hovering. (I suppose that is similar to saying that all countries in the top 5%-tile are deep red.)
> 
> This may need some experimentation (and we can change it any time, as the colour will change every day anyway), but I think is worth doing.
> 
> You could summarise my thoughts as: I think this should look more alarming than it is at the moment ... ;-)
> 
> I created an issue for this (#199).

Changed it now so that it goes up to the 90th percentile, and added an explanation above the plot stating what that means, not too sure if it makes sense though

fangohr commented:
Very cool - the 90-percentile plot looks great. 

I also like the "The colour scale on this world map goes up to the 90th quantile (where 90% of the countries have less than that number of cases), any country with a 7 day incidence rate over the 90th quantile is set to the darkest red colour." text, although I would put it below the figure.

How about this caption (to go below the nice world-map figure):

> The map shows 7-day incidence rates per 100,000 people based on data reported by the Johns-Hopkins University. The colour scale for the this world map goes up to the 90th quantile (where 90% of the countries have less than that number of cases): any country with a 7 day incidence rate over the 90th quantile is set to the darkest red colour."

fangohr commented:
Could we change the label above the colour bar from "7 Day Incidence Rate" to "7-day incidence/100k"? That's not longer but gets the population normalisation in.

RobertRosca commented:
TODO: remove everything apart from + - and reset from the `ModeBar`

RobertRosca commented:
Implemented requested changes and rebased from master so that the CSS changes are included in the preview.
